nntrainer/graph/network_graph.cpp

```diff
@@ -795,7 +796,12 @@ void NetworkGraph::inPlaceOptimize(const std::string &layer_type,
                                    Manager &manager) {
   for (auto &layer_node : getSorted()) {
     auto &l = layer_node.getObject();
-    if (l->getType() == layer_type &&
+    std::string l_type = l->getType();
```

Comment:
Please add a note here to verify if time distributed layers can work in-place. This will need checking and some unittests.

Reply:
Surely it needs unittest. For now, only two unittests are. I will add more in other PR including inPlaceOptimization. Talking about the in place opt, there is no reason not to work if the distributed layer is in place layer.

Reply:
> Talking about the in place opt, there is no reason not to work if the distributed layer is in place layer.

I agree, but this is just for safety :)